lib/tagged_content_exporter.rb

```diff
@@ -0,0 +1,72 @@
+class TaggedContentExporter
+  TRANSPORT_TAXON_ID = "a4038b29-b332-4f13-98b1-1c9709e216bc".freeze
+
+  def self.call
+    new.content_items_with_taxons.compact
+  end
+
+  def initialize
+    @content_items = ProjectContentItem
+      .for_taxonomy_branch(TRANSPORT_TAXON_ID)
+      .done
+  end
+
+  def content_items_with_taxons
+    @content_items.map do |content_item|
+      {
+        content_id: content_item.content_id,
+        url: content_item.url.gsub("https://www.gov.uk", ""),
+        taxons: taxons_for(content_item).compact
+      }
+    end
+  end
+
+private
+
+  def taxons_for(content_item)
+    links = Services.publishing_api.get_links(content_item.content_id).to_h
+    (links.dig("links", "taxons") || []).map do |taxon_id|
+      {
+        content_id: taxon_id
+      }.merge(metadata_for_taxon(taxon_id, content_item.content_id))
+    end
+  end
+
+  def metadata_for_taxon(taxon_id, content_id)
+    metadata = taxon_links_change(taxon_id, content_id)
+
+    user = User.find_by(uid: metadata.dig("user_uid")) || User.new
```

Comment:
The `User.new` bit here looks a bit odd to me.

Another option might be to not do that, and use `user.try(:organisation_slug)` below?

Reply:
This was added as a lazy implementation for the null pattern (e.g. instead of a `NullUser` object, perhaps). I think originally we were also taking out the `uid` from the user, so following this pattern was more beneficial before hand rather than scattering `try` around.